src/ur/dashboard_client_implementation_g5.cpp

```diff
@@ -0,0 +1,1195 @@
+    {
+      response.ok = false;
+      response.message = "Failed to release brakes. Expected 'robotmode' to be 'RUNNING', but did not receive it.";
+    }
+  }
+  catch (const UnexpectedResponse& e)
+  {
+    response.message = e.what();
+    response.ok = false;
+  }
+
+  return response;
+}
+
+DashboardResponse DashboardClientImplG5::commandLoadProgram(const std::string& program_file_name)
+{
+  // We load the program, which will fail if the program is not found or the requested file does
+  // not contain a valid program. Afterwards, we wait until the program state is stopped with a
+  // program named the same as the requested program. We cannot check the full file path here, but
+  // the important thing is that the program state is stopped.
+  DashboardResponse response;
+  try
+  {
+    response.message =
+        sendRequestString("load", "(?:Loading program: ).*(?:" + program_file_name + ").*", program_file_name);
+    response.ok = true;
+    response.data["program_name"] = program_file_name;
+    if (!waitForReply("programState", "STOPPED " + std::filesystem::path{ program_file_name }.filename().string()))
+    {
+      response.ok = false;
+      response.message = "Failed to load program. Expected 'programState' to be 'STOPPED " +
+                         std::filesystem::path{ program_file_name }.filename().string() + "', but did not receive it.";
+    }
+  }
+  catch (const UnexpectedResponse& e)
+  {
+    response.ok = false;
+    response.message = e.what();
+  }
+  return response;
+}
+
+DashboardResponse DashboardClientImplG5::commandLoadInstallation(const std::string& installation_file_name)
+{
+  DashboardResponse response;
+  try
+  {
+    response.message =
+        sendRequestString("load installation", "(?:Loading installation: ).*(?:" + installation_file_name + ").*",
+                          installation_file_name);
+    response.ok = true;
+  }
+  catch (const UnexpectedResponse& e)
+  {
+    response.ok = false;
+    response.message = e.what();
+  }
+
+  return response;
+}
+
+DashboardResponse DashboardClientImplG5::commandPlay()
+{
+  DashboardResponse response;
+  try
+  {
+    response.message = sendRequestString("play");
+    response.ok = true;
+    if (!waitForReply("programState", "(?:PLAYING ).*"))
+    {
+      response.ok = false;
+      response.message = "Failed to play program. Expected 'programState' to be 'PLAYING', but did not receive such a  "
+                         "message.";
+    }
+  }
+  catch (const UnexpectedResponse& e)
+  {
+    response.ok = false;
+    response.message = e.what();
+  }
+
+  return response;
+}
+
+DashboardResponse DashboardClientImplG5::commandPause()
+{
+  DashboardResponse response;
+  try
+  {
+    response.message = sendRequestString("pause");
+    response.ok = true;
+    if (!waitForReply("programState", "(?:PAUSED ).*"))
+    {
+      response.message = "Failed to pause program. Expected 'programState' to be 'PAUSED', but didn't receive such a "
```

Comment:
Is response supposed to be false if program state doesn't get to PAUSED? That is how it is implemented for play.